Feature/#16 GitHub pull requests

markspanbroek opened the pull request:
Closes #16 

Added Smart Contract for registering GitHub pull requests. The contract requests merged status and creator from the GitHub API using Oraclize. Unit tests and integration test are provided.

This pull request includes an update to the latest version of TruffleDevEnv.

svdo commented on src/app-contracts/contracts/PullRequests.sol:
```diff
@@ -0,0 +1,120 @@
+        repo = repo_;
+    }
+
+    function creators(uint pullRequestId) constant returns (string creator) {
```
function `creators` (plural) returns string `creator` (singular) ?

markspanbroek replied:
Yes, I tried to mimic the auto-generated getter function for mappings, which follow this pattern. But now that you mention it, it is confusing. I will change that.

markspanbroek commented on src/app-contracts/test/test-networks.js:
```diff
@@ -4,8 +4,16 @@ const testNetworks = {
 
-function isTestNetwork (web3) {
-  return !!testNetworks[web3.version.network]
+async function isTestNetwork (web3) {
```
That is technically speaking true, but I chose to mark it as async anyway :) The code returns a promise, therefore callers should be aware that it's an async function. Now they can see that by looking at the function signature.

markspanbroek commented:
Regarding validation failures: I always like to be explicit about them, therefore *throw*. Also to ensure that there are no side-effects.

markspanbroek commented:
@svdo I renamed the function `creators` to `creator`. I also changed `users` to `user` in the Users contract.

Additionally, I updated to the latest version of TruffleDevEnv to support the latest yarn and node. You might need to update your yarn and node as well.
